File: web-app-react-admin/src/models/product.ts
import { productQuantityVM } from "./productQuantity";

export interface ProductVM {
    id?: number
    name?: string
    title?: string
    image?: string
    price?: number
    code?: string
    originalPrice?: number
    description?: string
    content?: string
    ViewCount?: number
    tags?: string
    seoAlias?: string
    seoKeywords?: string
    seoDescription?: string
    productCategoryId?: number
    productCategoryName?: number
    status?: number
    productQuantity?: productQuantityVM[]
    productImages?: ProductImageVM[]
}

export interface ProductImageVM {
    id: number
    productId: number
    fileName: string
    path: string
    size: number
    extension: string
    type: string
    sortOrder: number
    caption: string
    createdBy: string
    updatedBy: string
    dateCreated: Date
    dateModified: Date
}

File: WebAppIdentityServer/src/WebAppIdentityServer.Data.EF/Entities/Product.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using WebAppIdentityServer.Data.EF.Interfaces;
using WebAppIdentityServer.Utilities.Enum;

namespace WebAppIdentityServer.Data.EF.Entities
{
    public class Product : IDateTracking, IEntityTracking, IHasSeoMetaData
    {
        public long Id { get; set; }
        public string Name { get; set; }
        public string Title { get; set; }
        public string Image { get; set; }
        public decimal Price { get; set; }
        public decimal OriginalPrice { get; set; }
        public string Description { get; set; }
        public string Content { get; set; }
        public string Code { get; set; }
        public int? ViewCount { get; set; }
        [StringLength(255)]
        public string Tags { get; set; }
        public string SeoAlias { get; set; }
        public string SeoKeywords { get; set; }
        public string SeoDescription { get; set; }
        public long ProductCategoryId { set; get; }
        public ProductCategory ProductCategory { set; get; }
        public IEnumerable<ProductImages> ProductImages { set; get; }
        public IEnumerable<ProductTag> ProductTags { set; get; }
        public IEnumerable<ProductQuantity> ProductQuantity { set; get; }
        public string CreatedBy { get; set; }
        public string UpdatedBy { get; set; }
        public DateTime DateCreated { get; set; }
        public DateTime? DateModified { get; set; }
        public Status Status { get; set; }
    }
}
